consensus/src/payload_manager.rs

```diff
@@ -181,7 +186,17 @@ impl PayloadManager {
                                 },
                             }
                         }
-                        let ret: Vec<SignedTransaction> = vec_ret.into_iter().flatten().collect();
+                        let ret: Vec<OrderedSignedUserTransaction> = vec_ret
+                            .into_iter()
+                            .enumerate()
+                            .map(|(batch_index, txns)|
+                                txns.into_iter().map(|t| OrderedSignedUserTransaction{
+                                    transaction: t,
+                                    batch_index: u16::try_from(batch_index).unwrap(),
```

Comment:
i think it can be just batch_index as u16, it doesn't panic if it overflows but we probably don't expect it overflows

Reply:
hmm... TIL this isn't counted as overflowing .. in that case I think an explicit panic is proper :P